### PR TITLE
changelog app: repo updates

### PR DESCRIPTION
**Pull request by aerickson**
- changelog
  - ronin-puppet is the new repo that is replacing build-puppet
  - relops now uses a fork of bc's devicepool repo
- pre-commit
  - relax Black version requirement to any python3 (vs only 3.7)
    - field is used to locate the binary to use, see https://black.readthedocs.io/en/stable/version_control_integration.html

### CODE REVIEW

**Comment by aerickson** [comment]
cc @gmierz @davehunt

**Comment by aerickson** [comment]
@ionutgoldan, any thoughts? Thanks. :)

**Comment by ionutgoldan** [comment]
Hmm... I'm not too familiar with the changelog service.
I promise that by tomorrow I'll test this PR & give a resolution.

File: treeherder/changelog/repositories.json
Comment (ionutgoldan) [inline treeherder/changelog/repositories.json:38]: I get a 404 from Github for this repo.

**Review by ionutgoldan — CHANGES_REQUESTED** [review]

File: treeherder/changelog/repositories.json
Comment (aerickson) [reply treeherder/changelog/repositories.json]: Fixed.

**Review by ionutgoldan — APPROVED** [review]
The registration looks good.

**Comment by ionutgoldan** [comment]
@sarah-clements could you look over the CircleCI & pre-commit configs?

File: .circleci/config.yml
Comment (sarah-clements) [inline .circleci/config.yml:17]: Is the urlchecker validating the json file? I'd rather extend any sort of validating or linting to all .json files rather than one specific file (and ideally to toml and yml files too). I'm thinking this might be better done as part of the linter checks.

File: .pre-commit-config.yaml
Comment (sarah-clements) [inline .pre-commit-config.yaml:2]: Thanks for adding this check but the url is wrong :)

Comment (aerickson) [reply .pre-commit-config.yaml:2]: Can you expand on how it's wrong? Does treeherder maintain it's own fork of these or?

`pre-commit` runs and is happy with the URL.

File: .circleci/config.yml
Comment (aerickson) [reply .circleci/config.yml:17]: Yes, only the json files in the changelog subdirectory. I pushed an earlier commit with a bad URL and wanted to make sure it wouldn't happen again.

I'm happy to extend it to everything in the repo, but didn't want to decide that myself.

By linter, you mean eslint?

File: .pre-commit-config.yaml
Comment (sarah-clements) [reply .pre-commit-config.yaml:2]: Apologies, user error on my part - I must have made a type-o when I took a look at it.

File: .circleci/config.yml
Comment (sarah-clements) [reply .circleci/config.yml:17]: yes, eslint (or an eslint plugin) - thank you!

Comment (aerickson) [reply .circleci/config.yml:17]: ESLint seems focused on JS files. It didn't like the syntax for repositories.json or the other existing JSON files (I also couldn't find a link checker plugin).

modified `yarn lint` command to add JSON:
``` bash
$ node ./node_modules/eslint/bin/eslint.js --report-unused-disable-directives --max-warnings 0 --format codeframe --ext js,jsx,json  treeherder/changelog/repositories.json 
error: Parsing error: Missing semicolon

  1 | {
> 2 |   "repositories": [
    |                 ^
  3 |     {
  4 |       "metadata": {
  5 |         "description": "Tool for creating Windows cloud", at treeherder/changelog/repositories.json:2:17:
  1 | {
> 2 |   "repositories": [
    |                 ^
  3 |     {
  4 |       "metadata": {
  5 |         "description": "Tool for creating Windows cloud",


1 error found.
```

Should I just abandon the check, create an urlchecker pre-commit hook (should be easy), or do you have any other ideas? Thanks. :)

Comment (sarah-clements) [reply .circleci/config.yml:17]: Ah, darn. The urlchecker pre-commit hook sounds good.

**Comment by aerickson** [comment]
@sarah-clements I've got this working. Just checking MD and JSON files for now (ignoring node_modules and tests also). There are a few files with bad URLs... if you let me know what you'd like done with them I'm happy to do it. I can ignore URLs or certain files/paths in the pre-commit config.

Failing URLs (should be the same as the failing CircleCI builds link above):
https://app.circleci.com/pipelines/github/mozilla/treeherder/1443/workflows/f16baa89-3080-40cf-9405-74fa0ea89f81/jobs/4968

I'm still working with the upstream project to get these changes committed and released (it could be awhile), but the code works fine from my fork (or I can fork it into our mozilla-platform-ops org). Would you like to wait for upstream to merge or are you OK with a fork?

**Comment by sarah-clements** [comment]
> @sarah-clements I've got this working. Just checking MD and JSON files for now (ignoring node_modules and tests also). There are a few files with bad URLs... if you let me know what you'd like done with them I'm happy to do it. I can ignore URLs or certain files/paths in the pre-commit config.
> 

You can ignore them and I'll deal them up - you've done enough to get this set up!

> I'm still working with the upstream project to get these changes committed and released (it could be awhile), but the code works fine from my fork (or I can fork it into our mozilla-platform-ops org). Would you like to wait for upstream to merge or are you OK with a fork?

When you say upstream project do you mean the ronin-puppet or the fork of bc's devicepool repo? Not sure I follow...

**Comment by aerickson** [comment]
> > @sarah-clements I've got this working. Just checking MD and JSON files for now (ignoring node_modules and tests also). There are a few files with bad URLs... if you let me know what you'd like done with them I'm happy to do it. I can ignore URLs or certain files/paths in the pre-commit config.
> 
> You can ignore them and I'll deal them up - you've done enough to get this set up!

The builds step in CI is failing because tox runs `pre-commit run -a`. Are you OK landing with it broken or  should I move the new check into a separate PR?

> > I'm still working with the upstream project to get these changes committed and released (it could be awhile), but the code works fine from my fork (or I can fork it into our mozilla-platform-ops org). Would you like to wait for upstream to merge or are you OK with a fork?
> 
> When you say upstream project do you mean the ronin-puppet or the fork of bc's devicepool repo? Not sure I follow...

The tool I'm using for checking URLs (https://github.com/urlstechie/urlchecker-python) didn't support the way pre-commit invokes tools, so I had to modify it slightly. pre-commit will run from a fork (how it's working in this PR) or we can wait until the code is merged and released in the official repo (I'm working with the authors, but not sure when that might be).

**Comment by sarah-clements** [comment]
> The builds step in CI is failing because tox runs `pre-commit run -a`. Are you OK landing with it broken or should I move the new check into a separate PR?
> Let's move it into another pr.

> The tool I'm using for checking URLs (https://github.com/urlstechie/urlchecker-python) didn't support the way pre-commit invokes tools, so I had to modify it slightly. pre-commit will run from a fork (how it's working in this PR) or we can wait until the code is merged and released in the official repo (I'm working with the authors, but not sure when that might be).

I'd be ok waiting but I don't know how quickly you need this pr to merge. If it can wait a week, let's do that. If it needs to be merged tomorrow let's use the fork.

**Comment by aerickson** [comment]
All but one of the pre-commit changes have been moved to #7166. 

The one remaining change (changing the executable used to run black), is required for me to run pre-commit on my system. I only have py3.9. I think `python3` is what we want as black just requires a valid py3.

**Comment by aerickson** [comment]
> All but one of the pre-commit changes have been moved to #7166.
> 
> The one remaining change (changing the executable used to run black), is required for me to run pre-commit on my system. I only have py3.9. I think `python3` is what we want as black just requires a valid py3.

Everything has been moved to #7166 to make this PR as simple as possible.

**Review by sarah-clements — APPROVED** [review]

**Pull request merged**
